inst/include/cpp11/doubles.hpp

```diff
@@ -134,33 +134,44 @@ typedef r_vector<double> doubles;
 
 }  // namespace writable
 
+template <>
+inline double na() {
+  return NA_REAL;
+}
+
+template <>
+inline bool is_na(const double& x) {
+  return ISNA(x);
+}
+
+// forward declarations
 typedef r_vector<int> integers;
 
+template <>
+int na();
+
+template <>
+int r_vector<int>::operator[](const R_xlen_t pos) const;
+
 inline doubles as_doubles(sexp x) {
   if (TYPEOF(x) == REALSXP) {
     return as_cpp<doubles>(x);
-  }
-
-  else if (TYPEOF(x) == INTSXP) {
+  } else if (TYPEOF(x) == INTSXP) {
     integers xn = as_cpp<integers>(x);
-    size_t len = xn.size();
-    writable::doubles ret;
-    for (size_t i = 0; i < len; ++i) {
-      ret.push_back(static_cast<double>(xn[i]));
+    R_xlen_t len = xn.size();
```

Comment:
I think `R_xlen_t` is better than `size_t` in this case, since:
- `.size()` returns an `R_xlen_t`
- The doubles constructor takes an `R_xlen_t`
- The `[i]` method takes a `R_xlen_t` `i` value